apis/python/src/tiledbsoma/_tdb_handles.py

```diff
@@ -247,6 +313,104 @@ def _do_initial_reads(self, reader: tiledb.Group) -> None:
         }
 
 
+class DataFrameWrapper(Wrapper[clib.SOMADataFrame]):
```

Comment:
since this is for reads only, would it make sense to call this a DataFrameReadWrapper?

Reply:
I am going to change the write path to use `DataFrameWrapper` too in a future PR.